### PR TITLE
Use --show-iface-abi-hash for extracting hash

### PR DESCRIPTION
**Pull request by wavewave**
Use efficient new CLI API.

### CODE REVIEW

File: haskell/tools/ghc_wrapper.py [hunk 1]
```diff
@@ -131,13 +131,12 @@ def recompute_abi_hash(ghc, abi_out):
...
-            hash = line.split(":", 1)[1]
-            with open(abi_out, "w") as outfile:
-                print(hash, file=outfile)
-            return
+    cmd = [ghc, "--show-iface-abi-hash", hi_file]
+    for line in subprocess.check_output(cmd, text=True):
+        hash = line[0]
+        with open(abi_out, "w") as outfile:
+            print(hash, file=outfile)
+        return
```
Comment (avdv) [inline haskell/tools/ghc_wrapper.py:139]: This only extracts the first character of the first line.

```suggestion
    return subprocess.check_output(cmd, text=True).split(maxsplit=1)[0]
```

Comment (avdv) [reply haskell/tools/ghc_wrapper.py:139]: Also, when testing that on my machine, I get diagnostic output mangled with the ABI hash:
```console
$ /nix/store/4kafspbkhbf590yd41m8d4m77j06j788-ghc-9.8.2-with-packages/bin/ghc --show-iface-abi-hash buck-out/v2/gen/root/904931f735703749/backend/__backend_infra__/mod-static/src/App.hi | bat
───────┬─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ STDIN
───────┼─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ Loaded package environment from /home/claudio/.ghc/x86_64-linux-9.8.2/environments/default
   2   │ 39589f7ccebb28f6f5b2130cfe55b660
   3   │ 
───────┴─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

```
Using `-v0` suppresses that.

Comment (wavewave) [reply haskell/tools/ghc_wrapper.py:139]: ah. 🤦 thanks for the catch!

Comment (wavewave) [reply haskell/tools/ghc_wrapper.py]: and yeah. GHC should have a clear separation of supplementary information/logging and intended output in terms of `stderr` and `stdout`.

**Comment by wavewave** [comment]
@avdv Thanks!

**Pull request closed without merging**